core/src/data/key.rs

```diff
@@ -130,12 +132,24 @@ impl TryFrom<Value> for Key {
             Uuid(v) => Ok(Key::Uuid(v)),
             Null => Ok(Key::None),
             Map(_) => Err(KeyError::MapTypeKeyNotSupported.into()),
-            List(_) => Err(KeyError::ListTypeKeyNotSupported.into()),
+            List(values) => Key::try_from(values),
             Point(_) => Err(KeyError::PointTypeKeyNotSupported.into()),
         }
     }
 }
 
+impl TryFrom<Vec<Value>> for Key {
+    type Error = Error;
+
+    fn try_from(values: Vec<Value>) -> Result<Self> {
+        values
+            .into_iter()
+            .map(Key::try_from)
+            .collect::<Result<Vec<_>>>()
+            .map(Key::List)
+    }
+}
+
```

Comment:
it would be not to have TryFrom. we already have too many From, TryFrom impls. it makes code harder to read but I didn't know that at the point I firstly add these.
currently, we are trying to use less From or TryFrom as possible.
this `TryFrom<Vec<Value>> for Key` is only being used in the above.
we can simply replace it.
```rust
List(values) => values
    .into_iter()
    .map(Key::try_from)
    .collect::<Result<Vec<_>>>()
    .map(Key::List),
```

Reply:
Resolved in commit b7f631b